intranet/apps/signage/views.py

```diff
@@ -26,3 +31,95 @@ def signage_display(request, display_id):
     context["sign"] = sign
     context["page_args"] = (sign, request)
     return render(request, "signage/base.html", context)
+
+
+def eighth(request):
+    block_id = request.GET.get("block_id")
+    block_increment = 0
+    internal_ip = check_internal_ip(request)
+    if internal_ip:
+        return internal_ip
+
+    if block_id is None:
+        block = EighthBlock.objects.get_first_upcoming_block()
+        if block is None:
+            last_block = EighthBlock.objects.order_by("date").last()
+            if last_block is not None:
+                block = last_block
+    else:
+        try:
+            block = (EighthBlock.objects.prefetch_related("eighthscheduledactivity_set").get(id=block_id))
+        except EighthBlock.DoesNotExist:
+            if EighthBlock.objects.count() == 0:
```

Comment:
Isn't this condition basically going to be never met unless we recreate all of Ion? I think this might be better if we check if the number of eighth blocks for this year is 0.
Also this probably should use `exists`